sdk/src/test/java/io/dapr/resiliency/RetryPolicyTest.java

```diff
@@ -19,14 +19,11 @@
 import org.junit.jupiter.api.Test;
 import reactor.core.Exceptions;
 import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
 
 import java.util.concurrent.atomic.AtomicInteger;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertSame;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.fail;
 
 public class RetryPolicyTest {
```

Comment:
Just did some clean up by using StepVerifier instead - it seems to be better practice but did not affect the fix. On the other hand, the hang reproduces as a timeout and test failure instead of hanging forever.